chains/solana/contracts/tests/ccip/ccip_router_test.go

```diff
@@ -9506,6 +9506,189 @@ func TestCCIPRouter(t *testing.T) {
+				)
+				instruction, err = raw.ValidateAndBuild()
+				require.NoError(t, err)
+
+				testutils.SendAndFailWithRPCError(ctx, t, solanaGoClient, []solana.Instruction{instruction}, transmitter, config.DefaultCommitment, []string{"VersionedTransaction too large"})
+
+				// We now build a buffer for the report.
+
+				bufferPDA, _, _ := solana.FindProgramAddress([][]byte{[]byte("execution_report_buffer"), root[:], transmitter.PublicKey().Bytes()}, config.CcipOfframpProgram)
+
+				// Deliberately chosen not a divider of 1000, to prove that a smaller terminator chunk is accepted.
+				const chunkSize = 300
+				// Arbitrary order starting on the buffer terminator, for maximum complexity.
+				order := []int{3, 1, 2, 0}
+
+				for _, i := range order {
+					start := i * chunkSize
+					end := min(((i + 1) * chunkSize), len(rawReport))
+
+					chunk := rawReport[start:end]
+
+					t.Log("Data: ")
+					t.Log(root[:])
+					t.Log(uint32(len(rawReport)))
+					appendIx, appendErr := ccip_offramp.NewBufferExecutionReportInstruction(
+						root[:],
+						uint32(len(rawReport)),
+						chunk,
+						uint8(i),
+						bufferPDA,
+						config.OfframpConfigPDA,
+						transmitter.PublicKey(),
+						solana.SystemProgramID).ValidateAndBuild()
+
+					require.NoError(t, appendErr)
+					testutils.SendAndConfirm(ctx, t, solanaGoClient, []solana.Instruction{appendIx}, transmitter, config.DefaultCommitment)
+				}
+
+				raw2 := ccip_offramp.NewExecuteInstruction(
+					// Empty report to execute from the buffer
+					[]byte{},
+					reportContext,
+					[]byte{},
+					config.OfframpConfigPDA,
+					config.OfframpReferenceAddressesPDA,
+					config.OfframpEvmSourceChainPDA,
+					rootPDA,
+					config.CcipOfframpProgram,
+					config.AllowedOfframpEvmPDA,
+					transmitter.PublicKey(),
+					solana.SystemProgramID,
+					solana.SysVarInstructionsPubkey,
+					config.RMNRemoteProgram,
+					config.RMNRemoteCursesPDA,
+					config.RMNRemoteConfigPDA,
+				)
+
+				raw2.AccountMetaSlice = append(
+					raw2.AccountMetaSlice,
+					solana.NewAccountMeta(config.CcipLogicReceiver, false, false),
+					solana.NewAccountMeta(config.OfframpReceiverExternalExecPDA, false, false),
+					solana.NewAccountMeta(config.ReceiverExternalExecutionConfigPDA, true, false),
+					solana.NewAccountMeta(config.ReceiverTargetAccountPDA, true, false),
+					solana.NewAccountMeta(solana.SystemProgramID, false, false),
+					// PDA of the buffer account
+					solana.NewAccountMeta(bufferPDA, true, false),
+				)
+				bufferedExecutionIx, err := raw2.ValidateAndBuild()
+				require.NoError(t, err)
+				tx = testutils.SendAndConfirm(ctx, t, solanaGoClient, []solana.Instruction{bufferedExecutionIx}, transmitter, config.DefaultCommitment, common.AddComputeUnitLimit(1000000))
+
+				executionEvents, err2 := common.ParseMultipleEvents[ccip.EventExecutionStateChanged](tx.Meta.LogMessages, "ExecutionStateChanged", config.PrintEvents)
+				require.NoError(t, err2)
+
```

Comment:
Could you please also add a check that the transmitter has recovered the SOL from the buffer rent after the fact? I get that anchor should handle it, but it's a relevant thing to guarantee so that NOPs don't leak funds when doing the buffered exec